scripts/zones/Windurst_Walls/npcs/Arbitrix.lua

```diff
@@ -7,47 +7,210 @@
+local adoulinOptionOff = 0x80
+local pictlogicaOptionOff = 0x100
+local wantedOptionOff = 0x1000
+local hideOptionFlags = adoulinOptionOff + pictlogicaOptionOff + wantedOptionOff
 
 function onTrigger(player, npc)
     local playerAgeDays = (os.time() - player:getTimeCreated()) / 86400
-
     local dailyTallyPoints = player:getCurrency("daily_tally")
     local firstVisit = dailyTallyPoints == -1
-
     local gobbieBoxUsed = player:getCharVar("gobbieBoxUsed")
-    local specialDialUsed = 1
-    local adoulinDialUsed = 1
-    local pictlogicaDialUsed = 1
-    local wantedDialUsed = 1
-
-    local adoulinOptionOff = 0x80
-    local pictlogicaOptionOff = 0x100
-    local wantedOptionOff = 0x1000
-    local hideOptionFlags = adoulinOptionOff + pictlogicaOptionOff + wantedOptionOff
+    local specialDialUsed = player:getMaskBit(gobbieBoxUsed, 0) and 1 or 0
+    local adoulinDialUsed = player:getMaskBit(gobbieBoxUsed, 1) and 1 or 0
+    local pictlogicaDialUsed = player:getMaskBit(gobbieBoxUsed, 2) and 1 or 0
+    local wantedDialUsed = player:getMaskBit(gobbieBoxUsed, 3) and 1 or 0
+    local holdingItem = player:getCharVar("gobbieBoxHoldingItem")
 
     if playerAgeDays >= GOBBIE_BOX_MIN_AGE and firstVisit then
         player:startEvent(519)
     elseif playerAgeDays >= GOBBIE_BOX_MIN_AGE then
-        player:startEvent(520, specialDialUsed, adoulinDialUsed, pictlogicaDialUsed, wantedDialUsed, 0, 0, hideOptionFlags, dailyTallyPoints)
+        if holdingItem ~= 0 then
+            player:startEvent(521)
+        else
+            player:startEvent(520, specialDialUsed, adoulinDialUsed, pictlogicaDialUsed, wantedDialUsed, 0, 0, hideOptionFlags, dailyTallyPoints)
+        end
     else
         player:messageSpecial(ID.text.YOU_MUST_WAIT_ANOTHER_N_DAYS, GOBBIE_BOX_MIN_AGE - playerAgeDays)
     end
 end
 
 function onEventUpdate(player, csid, option)
     local dailyTallyPoints = player:getCurrency("daily_tally")
+    local holdingItem = player:getCharVar("gobbieBoxHoldingItem")
+    local gobbieBoxUsed = player:getCharVar("gobbieBoxUsed")
+    local specialDialUsed = player:getMaskBit(gobbieBoxUsed, 0) and 1 or 0
+    local adoulinDialUsed = player:getMaskBit(gobbieBoxUsed, 1) and 1 or 0
+    local pictlogicaDialUsed = player:getMaskBit(gobbieBoxUsed, 2) and 1 or 0
+    local wantedDialUsed = player:getMaskBit(gobbieBoxUsed, 3) and 1 or 0
+    local itemid = 0
 
-    if csid == 520 and option == 4 then -- Peek items
-        player:updateEvent(200, 200, 200)
-    elseif csid == 520 and option == 9 then
-        player:updateEvent(1, 1, (dailyTallyPoints >= 10) or 1)
-    elseif csid == 520 and option == 10 then
-        -- Deliver Dial 1
+    if csid == 520 then
+        switch (option): caseof
+        {
+            [4] = function() -- Peek items
+                player:updateEvent(player:selectDailyItem(6), player:selectDailyItem(6), player:selectDailyItem(6))
+            end,
+            [9] = function() -- Dial 1
```

Comment:
Can shrink this option logic with the following:
```lua
if option == 4 then
    -- peek
else
    local dial = math.floor(option / 8)
    local option_type = option % 8
    local dial_used = 0
    local dial_cost = costs[dial]
    local dial_mask = false
    if dial >= 6 then
        dial_mask = dial - 6
        dial_used = player:getMaskBit(gobbieBoxUsed, dial_mask)
    end
    switch (option_type): caseof
        [1] = function()
            if dial_used then
                player:updateEvent(1, dial, 2) -- already used this dial
            elseif dailyTallyPoints >= dial_cost then
                itemid = player:selectDailyItem(dial)
                player:setCharVar("gobbieBoxHoldingItem", itemid)
                player:setCurrency("daily_tally", dailyTallyPoints - dial_cost)
                if dial_mask then
                    player:setMaskBit(gobbieBoxUsed, "gobbieBoxUsed", dial_mask, true)
                end
                player:updateEvent(itemid, dial, 0)
            else
                player:updateEvent(1, dial, 1) -- not enough points
            end
        end,
        [2] = function()
            if player:getFreeSlotsCount() == 0 then
                player:updateEvent(holdingItem, 0, 0, 1) -- inventory full, exit event
                player:messageSpecial(ID.text.ITEM_CANNOT_BE_OBTAINED + 2) -- generic "Cannot obtain the item."
            end
        end,
        [5] = function()
            if holdingItem > 0 and npcUtil.giveItem(player, holdingItem) then
                player:setCharVar("gobbieBoxHoldingItem", 0)
            end
            player:updateEvent(specialDialUsed, adoulinDialUsed, pictlogicaDialUsed, wantedDialUsed, 0, 0, hideOptionFlags, dailyTallyPoints)
        end,
    end
end
```

Will need a `costs` table somewhere. But in Arbitrix is okay - you don't need to globalize this just yet!

Reply:
This looks a lot better. I kinda just threw this together to demonstrate the item selection, but I'll make these edits and make sure everything works right.